### PR TITLE
[16.0][ADD] stock_available_immediately_exclude_location

### PR DESCRIPTION
**Pull request by glitchov**
Exclude location(s) from the computation of immediately usable quantity

### CODE REVIEW

File: stock_available_immediately_exclude_location/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+class ProductProduct(models.Model):
+
+    _inherit = "product.product"
+
+    def _compute_available_quantities_dict(self):
+        """
+        change the way immediately_usable_qty is computed by deducing the quants
+        in excluded locations
+        """
+        res, stock_dict = super(
```
Comment (rousseldenis) [inline stock_available_immediately_exclude_location/models/product_product.py]:
```suggestion
        res, stock_dict = super()._compute_available_quantities_dict()
```

File: stock_available_immediately_exclude_location/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        # of locations to exclude by adding the criteria on
+        # exclude_from_immediately_usable_qty and adapting the domain to work
+        # on stock.location.
+        # In this way we are always sure that the computation of qties to
+        # exclude is always done for locations part of the expected locations
+        # provided by the context/
+        # replace location_id by id into the quant domain to get the
+        # domain to apply to the locations
+        location_domain = []
+        for element in quant_domain:
```
Comment (rousseldenis) [inline stock_available_immediately_exclude_location/models/product_product.py:58]: I don't understand the need of this.

Comment (rousseldenis) [reply stock_available_immediately_exclude_location/models/product_product.py:58]: Ok, but I would have added a security feature to be sure no field from stock.quant will arrive in stock.location domain

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Review by lmignon — APPROVED** [review]
LGTM (Code review)

**Comment by lmignon** [comment]
@rousseldenis Could you update your review PLZ?

File: stock_available_immediately_exclude_location/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        # Be sure to exclude potential fields that couldn't belong to stock.location
+        # and replace such term by something True domain compatible
+        location_domain = []
+        location_fields = self.env["stock.location"].fields_get()
+        for element in quant_domain:
+            if expression.is_leaf(element) and element[0] not in location_fields:
+                # exclude the element from domain and replace by something True
+                location_domain.append((1, "=", 1))  # True and domain compatible
+            elif expression.is_leaf(element):
+                location_domain.append(
```
Comment (rousseldenis) [inline stock_available_immediately_exclude_location/models/product_product.py:63]: @glitchov This is weird that tests don't pass in this

Comment (rousseldenis) [reply stock_available_immediately_exclude_location/models/product_product.py:63]: @glitchov Could you check my comment ?

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
@rousseldenis I think this one is good. Can you update your review?

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**